src/test/java/com/auberer/compilerdesignlectureproject/parser/ForNodeTest.java

```
@@ -0,0 +1,76 @@
+package com.auberer.compilerdesignlectureproject.parser;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.auberer.compilerdesignlectureproject.ast.ASTForNode;
+import com.auberer.compilerdesignlectureproject.ast.ASTStmtLstNode;
+import com.auberer.compilerdesignlectureproject.lexer.Lexer;
+import com.auberer.compilerdesignlectureproject.lexer.TokenType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+public class ForNodeTest {
+
+    @Spy
+    private Parser parser;
+
+    @Mock
+    private Lexer lexer;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        parser = new Parser(lexer);
+        parser = spy(parser);
+    }
+
+    // Test fails, but we don't know why and can't fix it
+
+    @Test
+    @DisplayName("Test for node")
+    void testForNode() throws Exception {
```

Please catch all exceptions within the test body.